cmd/entrypoint/webhooks.go

```diff
@@ -0,0 +1,275 @@
+package entrypoint
+
+import (
+	"crypto/rsa"
+	"crypto/x509"
+	"crypto/rand"
+	"crypto/tls"
+	"crypto/x509/pkix"
+	"encoding/pem"
+	"math/big"
+	"net/http"
+	"context"
+	"time"
+	b "bytes"
+	"fmt"
+
+	"github.com/datawire/ambassador/v2/pkg/k8s"
+	"github.com/datawire/ambassador/v2/pkg/api/getambassador.io/v2"
+	"github.com/datawire/ambassador/v2/pkg/api/getambassador.io/v3alpha1"
+
+	k8sClientCoreV1 "k8s.io/client-go/kubernetes/typed/core/v1"
+	k8sTypesCoreV1 "k8s.io/api/core/v1"
+	k8sTypesMetaV1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	k8sErrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	clientgoscheme "k8s.io/client-go/kubernetes/scheme"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/conversion"
+	k8sApiExtClient "k8s.io/apiextensions-apiserver/pkg/client/clientset/clientset/typed/apiextensions/v1beta1"
+	k8sApiExtTypes "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1beta1"
+)
+
+const WEBHOOK_PORT int = 8043
+// TODO: automatic cert regeneration
+const CERT_VALID_DAYS int = 365
+const CA_SECRET_NAME string = "emissary-ingress-webhook-ca"
```

Comment:
Nit: Style: It's generally against Go style to have underscores in identifiers. Also, these are exported when they don't need to be.